src/main/scala/nl/knaw/dans/easy/multideposit/MultiDepositParser.scala

```diff
@@ -38,14 +39,17 @@ object MultiDepositParser extends DebugEnhancedLogging {
     debug(s"Start parsing SIP Instructions at $file")
 
     for {
-      content <- Try {
-        val rawContent = Source.fromFile(file)(Codec(encoding)).mkString
-        val parser = CSVParser.parse(rawContent, CSVFormat.RFC4180)
-        for {
-          record <- parser.getRecords.asScala
-          if record.size() > 0
-          if !record.get(0).isBlank
-        } yield record.asScala.toList
+      content <- {
+        managed(Source.fromFile(file)(encoding)).map(_.mkString)
+          .flatMap(rawContent => managed(CSVParser.parse(rawContent, CSVFormat.RFC4180)))
+          .map(parser => {
+            for {
+              record <- parser.getRecords.asScala
+              if record.size() > 0
+              if !record.get(0).isBlank
+            } yield record.asScala.toList
+          })
```

Comment:
Maybe this block should become  a function, because these nested for comprehensions look complicated